pkg/reconciler/v1alpha1/autoscaling/autoscaling.go

```diff
@@ -187,7 +187,8 @@ func (c *Reconciler) reconcile(ctx context.Context, key string, kpa *kpa.PodAuto
 
 	// Get the appropriate current scale from the metric, and right size
 	// the scaleTargetRef based on it.
-	if err := c.kpaScaler.Scale(ctx, kpa, metric.DesiredScale); err != nil {
+	appliedScale, err := c.kpaScaler.Scale(ctx, kpa, metric.DesiredScale)
```

Comment:
I proposed a nearly-identical signature change [here](https://github.com/knative/serving/pull/2244#discussion_r229766557) cc @markusthoemmes @josephburnett 

There is no place currently where we have:
 1. Desired Pods
 2. Observed Pods
 3. Ready Pods

If we can change this signature slightly more to return `(desiredScale int32, observedScale int32, error)` then combined with `got` below, we would have all three pieces of information here.

Reply:
Going to port this over to the referenced PR!

Reply:
@mattmoor I think for this PR it's fine as-is? I can add whatever metric I need on top in the reporting PR?

Reply:
SGTM